### ISSUE TITLE
conda cannot create an environment with only a constrained virtual package

### ISSUE THREAD

**Issue by jjhelmus** [conda/conda#10116]
## Current Behavior
Conda fails when asked to create a environment containing only a virtual package with a version constraint.  Adding an additional spec to the environment or removing the constraint works.

### Steps to Reproduce

```
$ conda create -n example __glibc --dry-run  # works
$ conda create -n example "__glibc>=0` openssl --dry-run # works
$ conda create -n example "__glibc>=0" --dry-run
Collecting package metadata (current_repodata.json): done        
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
                                                   
# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<
                                                   
    Traceback (most recent call last):                  
      File "/home/jhelmus/conda/lib/python3.7/site-packages/conda/exceptions.py", line 1079, in __call__              
        return func(*args, **kwargs)
      File "/home/jhelmus/conda/lib/python3.7/site-packages/conda/cli/main.py", line 84, in _main
        exit_code = do_call(args, p)                                                            
      File "/home/jhelmus/conda/lib/python3.7/site-packages/conda/cli/conda_argparse.py", line 82, in do_call
        return getattr(module, func_name)(args, parser)                                         
      File "/home/jhelmus/conda/lib/python3.7/site-packages/conda/cli/main_create.py", line 37, in execute
        install(args, parser, 'create') 
      File "/home/jhelmus/conda/lib/python3.7/site-packages/conda/cli/install.py", line 265, in install
        should_retry_solve=(_should_retry_unfrozen or repodata_fn != repodata_fns[-1]),
      File "/home/jhelmus/conda/lib/python3.7/site-packages/conda/core/solve.py", line 117, in solve_for_transaction
        should_retry_solve)
      File "/home/jhelmus/conda/lib/python3.7/site-packages/conda/core/solve.py", line 158, in solve_for_diff
        force_remove, should_retry_solve)
      File "/home/jhelmus/conda/lib/python3.7/site-packages/conda/core/solve.py", line 289, in solve_final_state
        self.determine_constricting_specs(spec, ssc.solution_precs)
      File "/home/jhelmus/conda/lib/python3.7/site-packages/conda/core/solve.py", line 315, in determine_constricting_specs
        if sp.name == spec.name][0]
    IndexError: list index out of range

`$ /home/jhelmus/conda/bin/conda create -n example __glibc>=0 --dry-run`

  environment variables:
                 CIO_TEST=<not set>
   CONDA_BACKUP_JAVA_HOME=
CONDA_BACKUP_JAVA_LD_LIBRARY_PATH=
        CONDA_DEFAULT_ENV=base
                CONDA_EXE=/home/jhelmus/conda/bin/conda
             CONDA_PREFIX=/home/jhelmus/conda
    CONDA_PROMPT_MODIFIER=
   CONDA_PYTHONBREAKPOINT=
         CONDA_PYTHON_EXE=/home/jhelmus/conda/bin/python
               CONDA_ROOT=/home/jhelmus/conda
              CONDA_SHLVL=1
     JAVA_LD_LIBRARY_PATH=/home/jhelmus/conda/jre/lib/amd64/server
                     PATH=/home/jhelmus/conda/bin:/home/jhelmus/bin:/home/jhelmus/conda/bin:/hom
                          e/jhelmus/conda/condabin:/home/jhelmus/bin:/usr/local/sbin:/usr/local/
                          bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bi
                          n
         PYTHONBREAKPOINT=ipdb.set_trace
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>
               WINDOWPATH=2

     active environment : base
    active env location : /home/jhelmus/conda
            shell level : 1
       user config file : /home/jhelmus/.condarc
 populated config files : /home/jhelmus/.condarc
          conda version : 4.8.3
    conda-build version : 3.18.11
         python version : 3.7.7.final.0
       virtual packages : __cuda=10.1
                          __glibc=2.27
       base environment : /home/jhelmus/conda  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/jhelmus/conda/pkgs
                          /home/jhelmus/.conda/pkgs
       envs directories : /home/jhelmus/conda/envs
                          /home/jhelmus/.conda/envs
               platform : linux-64
             user-agent : conda/4.8.3 requests/2.21.0 CPython/3.7.7 Linux/4.15.0-109-generic ubuntu/18.04.4 glibc/2.27
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False

```

## Expected Behavior
Provided that the virtual package spec is valid conda should create an empty environment in the same manner as when an un-restrained virtual package is specified.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
$ conda info

     active environment : base
    active env location : /home/jhelmus/conda
            shell level : 1
       user config file : /home/jhelmus/.condarc
 populated config files : /home/jhelmus/.condarc
          conda version : 4.8.3
    conda-build version : 3.18.11
         python version : 3.7.7.final.0
       virtual packages : __cuda=10.1
                          __glibc=2.27
       base environment : /home/jhelmus/conda  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/jhelmus/conda/pkgs
                          /home/jhelmus/.conda/pkgs
       envs directories : /home/jhelmus/conda/envs
                          /home/jhelmus/.conda/envs
               platform : linux-64
             user-agent : conda/4.8.3 requests/2.21.0 CPython/3.7.7 Linux/4.15.0-109-generic ubuntu/18.04.4 glibc/2.27
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
$ conda config --show-sources
==> /home/jhelmus/.condarc <==
add_pip_as_python_dependency: False
changeps1: False
ssl_verify: True
channels:
  - defaults
show_channel_urls: True
report_errors: False
```
</p></details>

**Comment by jjhelmus** [comment]
I'm curious if #10057 changes this behavior

**Comment by jjhelmus** [comment]
#10057 does not change this behavior.

**Comment by cjmartian** [comment]
@jjhelmus Is it safe to say that we can close this issue now?

**Comment by chenghlee** [comment]
@cjmartian: The fix is in PR #10138; we can close this ticket once AppVeyor gets unstuck and that PR is merged.

**Comment by chenghlee** [comment]
PR #10138 has been merged, and this issue seems to have been resolved:
```
$ conda info

     active environment : base
    active env location : /work/conda/devenv_linux-64
            shell level : 1
       user config file : /home/chl/.condarc
 populated config files : /home/chl/.condarc
          conda version : 4.8.4.post19+4d3fafc87
    conda-build version : 3.18.11
         python version : 3.8.5.final.0
       virtual packages : __cuda=10.2=0
                          __glibc=2.27=0
                          __archspec=1=x86_64
       base environment : /work/conda/devenv_linux-64  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /work/conda/devenv_linux-64/pkgs
                          /home/chl/.conda/pkgs
       envs directories : /work/conda/devenv_linux-64/envs
                          /home/chl/.conda/envs
               platform : linux-64
             user-agent : conda/4.8.4.post19+4d3fafc87 requests/2.24.0 CPython/3.8.5 Linux/5.4.0-42-generic ubuntu/18.04.5 glibc/2.27
                UID:GID : 1000:100
             netrc file : /home/chl/.netrc
           offline mode : False

$ conda create -n example "__glibc>=2" --dry-run
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /work/conda/devenv_linux-64/envs/example

  added / updated specs:
    - __glibc[version='>=2']

DryRunExit: Dry run. Exiting.
```
